agent-shell.el

```diff
@@ -389,6 +455,88 @@ Returns an empty string if no icon should be displayed."
   (interactive)
   (message "agent-shell v%s" agent-shell--version))
 
+;;; Extension API
+
+(defun agent-shell-get-state (&optional buffer)
```

Comment:
If it's an internal function, I'm generally ok with with &optional, but being a public function now it makes me a little more nervous to introduce additional params.

Mind switching to cl-defun and &key buffer? You'll still be able to invoke as `(agent-shell-get-state)` or `(agent-shell-get-state :buffer my-buffer)`.